plugins/modules/win_dsc.ps1

```diff
@@ -342,6 +342,32 @@ Function Invoke-SafeDscResource {
     }
 }
 
+Function Write-AnsibleError {
+    <#
+    .SYNOPSIS
+    Raises a standard terminating error.
+
+    .PARAMETER Msg
+    Error message text
+
+    .PARAMETER Failed
+    Whether or not the error triggers a failure. Defaults to '$true'.
+    #>
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $true)]
+        [String]$Msg,
+        [Boolean]$Failed = $true
```

Comment:
Being pedantic here this should really a switch, bool is useful on parameters if you have 3 states; true, false, and nothing whereas a switch is just on and off. But considering this is `Write-AnsibleError` then I don't see a need for having failed be optional at all here.

Reply:
Yea in this case there really isn't any reason, it can be added in the future if need be but these don't use `$module.FailJson/ExitJson` because the module object hasn't been initialised yet.